File: JavaAndScala/src/main/java/com/java/se/conclusion/design/factory/simple/common/laptop/Lenovo.java
package com.java.se.conclusion.design.factory.simple.common.laptop;

/**
 * 	This is a class to be used to test the implementation of simple factory pattern
 *  -- This class must implement a parent interface
 *  
 * @author VinceYuan
 *
 */
public class Lenovo implements Laptop {

	@Override
	public void startup() {
		System.out.println("A Lenovo laptop is starting up...");
	}

	@Override
	public void shutdown() {
		System.out.println("A Lenovo laptop is shutting down...");
	}
}

File: JavaAndScala/src/main/scala/org/lang/scala/conclusion/design/factory/simple/common/laptop/Lenovo.scala
package org.lang.scala.conclusion.design.factory.simple.common.laptop

import org.lang.scala.conclusion.design.factory.simple.common.laptop.parent.Laptop

/**
 * 	This is a stand-alone class to be used to test the implementation of simple factory pattern
 *  -- This class must implement a parent interface
 *  
 * 	@author VinceYuan
 */
class Lenovo extends Laptop {
  
  override def startup(): Unit = {
    println("A Lenovo laptop is starting up...")
  }
  
  override def shutdown(): Unit = {
    println("A Lenovo laptop is shutting down...")
  }
}